test/node-http-test.js

```diff
@@ -0,0 +1,283 @@
+        assert.equal(nameInfoAfter.info.name, NAME0);
+        assert.equal(nameInfoAfter.info.state, 'OPENING');
+      });
+    });
+
+    describe('For names that are available at height 20', () => {
+      it('It should getNameInfo for an opening name', async () => {
+        await mineBlocks(20, cbAddress);
+
+        await wclient.execute('sendopen', [NAME1]);
+        await mineBlocks(1, cbAddress);
+
+        const nameInfo = await nclient.get(`/name/${NAME1}`);
+        assert(nameInfo.start.start < 20);
+        assert.equal(nameInfo.start.reserved, false);
+        assert.equal(nameInfo.info.state, 'OPENING');
+      });
+    });
+  });
+
+  describe('getNameByHash', () => {
+    it('It should return null when an auction has not been initiated', async () => {
+      const nameHash = rules.hashName(NAME0);
+      const name = await nclient.get(`/resource/hash/${nameHash.toString('hex')}`);
+      assert.equal(name, null);
+    });
+
+    describe('When an auction has been initiated', () => {
+      it('It should return the name', async () => {
+        await mineBlocks(10, cbAddress);
+
+        await wclient.execute('sendopen', [NAME0]);
+        await mineBlocks(1, cbAddress);
+
+        const nameHash = rules.hashName(NAME0);
+        const { name } = await nclient.get(`/resource/hash/${nameHash.toString('hex')}`);
+        assert.equal(name, NAME0);
+      });
+    });
+  });
+
+  describe('getNameResource', () => {
+    const records = { compat: false, version: 0, ttl: 172800, ns: ['ns1.example.com.@1.2.3.4'] };
+    it('It should return null when an auction has not been initiated', async () => {
+      const resource = await nclient.get(`/resource/name/${NAME0}`);
+      assert.equal(resource, null);
+    });
+
+    describe('When an auction has been initiated', () => {
+      it('It should return the resource', async () => {
+        await mineBlocks(10, cbAddress);
+
+        await wclient.execute('sendopen', [NAME0]);
+        await mineBlocks(1, cbAddress);
+
+        const { stats: { blocksUntilBidding } } = await wclient.execute('getauctioninfo', [NAME0]);
+
+        await mineBlocks(blocksUntilBidding, cbAddress);
+
+        const sendBid = await wclient.execute('sendbid', [NAME0, 12, 12]);
+        assert(sendBid);
+
+        const { stats: { blocksUntilReveal } } = await wclient.execute('getauctioninfo', [NAME0]);
+
+        await mineBlocks(blocksUntilReveal, cbAddress);
+
+        await wclient.execute('sendreveal', [NAME0]);
+        const { stats: { blocksUntilClose } } = await wclient.execute('getauctioninfo', [NAME0]);
+
+        await mineBlocks(blocksUntilClose, cbAddress);
+
+        await wclient.execute('sendupdate', [NAME0, records]);
+        await mineBlocks(1, cbAddress);
+
+        const resource = await nclient.get(`/resource/name/${NAME0}`);
+        assert.deepEqual(resource, Object.assign(records, { name: NAME0, version: 0 }));
+      });
+    });
+  });
+
+  describe('getNameProof', () => {
```

Comment:
Do you think we could dedupe `getnameproof` and `getnameproofbyhash` with a loop?

Reply:
:thinking: Yess - we could. I don't think it would improve anything however. Do you feel strongly about this?

Reply:
There is solid overlap - but there's only "2 things" to abstract, this feels like a situation I'd want to hold off on abstracting into a loop. Your call. But I wanted to voice the thought :ok: